SCClassLibrary/JITLib/ProxySpace/wrapForNodeProxy.sc

```diff
@@ -296,10 +296,13 @@
 				};
 
 				{ | out |
-					var e = EnvGate.new * Control.names(["wet"++(index ? 0)]).kr(1.0);
+
+					var e, ctl = Control.names(["wet"++(index ? 0)]).kr(1.0);
 					if(proxy.rate === 'audio') {
+						e = ctl * EnvGate(i_level: 0, doneAction:2, curve:\sin);
 						XOut.ar(out, e, SynthDef.wrap(func, nil, [In.ar(out, proxy.numChannels)]))
 					} {
+						e = ctl * EnvGate(i_level: 0, doneAction:2, curve:\lin);
 						XOut.kr(out, e, SynthDef.wrap(func, nil, [In.kr(out, proxy.numChannels)]))				};
 				}.buildForProxy( proxy, channelOffset, index )
```

Comment:
We can get rid of repetitious code here as well.

```
{ | out |
    var in, sig;
    var ctl = Control.names(["wet" ++ (index ? 0)]).kr(1.0);
    var rate = if(proxy.rate === \audio) { \audio } { \control };
    var curve = switch(rate) { \audio } { \sin } { \lin };
    var xfade = ctl * EnvGate(i_level: 0, doneAction: 2, curve: curve);
    in = In.perform(
        In.methodSelectorForRate(rate),
        out,
        proxy.numChannels
    );
    sig = SynthDef.wrap(func, nil, [in]);
    XOut.ar(out, xfade, sig);
}.buildForProxy( proxy, channelOffset, index )
```

Reply:
Yes, thought about this, too. But in the end, I found the longer version more readable. Also you forgot that `XOut` also needs a separate rate in both cases, so it gets longer. 

As far as I remember, `In.methodSelectorForRate(rate),` is not needed, because we can just write `In.multiNew(rate, proxy.numChannels, out)`.

Still, I prefer the other version for readability.